loader.py

```diff
@@ -54,45 +57,57 @@ def start(conf, tags, type, startdate, file, noupload):
                     try:
                         jsonResponse = myResponse.json()
 
-                        for key in jsonResponse:
-                            val = jsonResponse[key]
-                            if "has_more" in key:
-                                index += 1
-                                if val is not True:
-                                    # We are done
-                                    finished = False
-                                    print("#### Finished #####")
-                                    break
-                                else:
-                                    if isinstance(myResponse.links, dict):
-                                        res = myResponse.links["next"]
-                                        url = res["url"]
-                                        print("#### Continue #####")
+                        # Check if there are more values
+                        if 'has_more' in jsonResponse:
+                            val = jsonResponse['has_more']
+                            if val is not True:
+                                finished = False
+                                logging.info("There are no more attributes")
+                                logging.info("#### Finished #####")
+                                break
                             else:
-                                if isinstance(val, list) and "params" not in key:
-                                    print("Parsing... - Index: " + str(index))
-                                    if type == 'c2server':
-                                        C2Server.parse(event, val, tags)
-                                    elif type == 'malware':
-                                        Malware.parse(event, val, tags)
+                                if isinstance(myResponse.links, dict):
+                                    res = myResponse.links["next"]
+                                    url = res["url"]
+                                    logging.info("#### Continue #####")
+                        if 'iocs' in jsonResponse:
+                            val = jsonResponse['iocs']
+                            logging.info("Parsing... - Offset: " + str(index) + " to " + str(index + len(val)))
+                            index += len(val)
+
+                            if type == 'c2server':
+                                C2Server.parse(event, val, tags)
+                            elif type == 'malware':
+                                Malware.parse(event, val, tags)
+                        else:
+                            logging.warning("TIE answered with an empty reply")
 
                     except ValueError:
-                        print("Error:")
-                        print("Invalid or empty JSON Response")
+                        logging.error("Error:\nInvalid or empty JSON Response")
```

Comment:
Nitpick: Why the multiline error message? The error level should already be defined by the choice of logging method.